docs/adrs/v3-setup-dotnet.md

```diff
@@ -0,0 +1,48 @@
+
+- The action contains logic to handle inputs with wildcards, for example `5.0.x`, `5.0.*`, `5.x` or `5.*`. This logic uses metadata from `https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json` to retrieve the list of available releases and get the latest release version for the specified major and/or minor version from input. After that, installer script (`dotnet-install.ps1` for Windows or `dotnet-install.sh` for Linux and MacOS) installs required SDK using exact version as a parameter.
+
+# Proposal
+- Change .NET installation path for Windows and Ubuntu images to match the location of pre-installed versions by using `-InstallDir` (Windows) and `--install-dir` (Ubuntu) property for installer scripts:
+https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script
+- Get rid of redundant logic to resolve wildcard version on Action side and start relying on [official installer scripts provided by .NET Core team](https://github.com/dotnet/install-scripts).  
+The execution `dotnet-install.ps1 -Channel 5.0` installs the latest patch version for 5.0 sdk. In this way we can handle inputs with wildcard as patch version (`5.0.x` or `5.0.*`) by passing major and minor version to installer script directly as `channel` parameter. This parameter supports two-part version in X.Y format as input values ([see installer scripts documentation](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script)). In further the version input format of action will follow the format which is supported by installers scripts. 
+
+# Breaking changes
+- The presence of pre-installed .NET versions that are higher than version that the users specify in the setup task can be breaking for some customers, who expect only one installed .NET version on runner after using setup task. If user doesn't have .NET version specified in project file, `dotnet` will use the latest installed version instead of provided in setup task.  
+> **Note:** It is the biggest deal in this ADR.
+Previously, when user specifies .NET version, this exact version was used by `dotnet` command by default (because it was installed in the separate folder and there are no any other .NET versions in that folder)  
+In proposal, the specified version will be installed on machine but the latest version will be used by `dotnet` command by default (because specified version will be installed alongside with pre-installed .NET versions).  
+Based on [official .NET documentation](https://docs.microsoft.com/en-us/dotnet/core/versions/selection), it is expected behavior and how it works on user's local machine with multiple installed .NET versions but some customers could be broken because they already rely on current behavior.
+
+- All possible inputs will continue to work as previously except `5.x`. It is totally okay to drop it because we shouldn't allow user to rely on such input. New minor release of .NET will break their builds
```

Comment:
- All possible inputs will continue to work as previously except `5.x`. It is totally okay to drop it because we shouldn't allow users to rely on such input. The new minor release of .NET will break their builds